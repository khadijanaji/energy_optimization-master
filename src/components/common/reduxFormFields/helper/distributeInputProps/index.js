/* eslint-disable react/display-name */
// @flow

import React from "react";
import {createMuiTheme, MuiThemeProvider} from "@material-ui/core/styles";
import green from "@material-ui/core/colors/green";


const theme = createMuiTheme({"palette": {"primary": green}});

export default (MyComponent: any) => ({
  input,
  value,
  "meta": {touched, invalid, error},
  ...others
}: any) => (<MuiThemeProvider theme={theme}>
  <MyComponent
    {...input}
    {...others}
    error={touched && (invalid || Boolean(error))}
    value={value}
  />
</MuiThemeProvider>);
