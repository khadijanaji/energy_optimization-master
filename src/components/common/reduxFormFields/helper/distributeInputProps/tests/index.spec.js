import {shallow} from "enzyme";
import React from "react";

import distributeInputProps from "../";

const props = {
  "input": {},
  "value": "Value",
  "meta": {
    "touched": true,
    "invalid": true,
    "error": true
  },
  "other1": "Other",
  "other2": {}
};

describe("Component", () => {
  describe("distributeInputProps Render", () => {
    it("Children contain passed props: Should match snapshot", () => {
      const component = distributeInputProps(<input />);
      const wrapper = shallow(<component {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
