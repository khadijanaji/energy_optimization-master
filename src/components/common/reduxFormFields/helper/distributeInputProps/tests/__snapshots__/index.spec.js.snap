// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component distributeInputProps Render Children contain passed props: Should match snapshot 1`] = `
<component
  input={Object {}}
  meta={
    Object {
      "error": true,
      "invalid": true,
      "touched": true,
    }
  }
  other1="Other"
  other2={Object {}}
  value="Value"
/>
`;
